sufia-models/app/models/concerns/sufia/collection_behavior.rb

```diff
@@ -15,10 +15,39 @@ def update_permissions
       self.visibility = "open"
     end
 
-    # Compute the sum of each file in the collection
+    # Compute the sum of each file in the collection using Solr to
+    # avoid having to proces GBs of data for a simple number. Can be
+    # cleaned up once it is confirmed to work
+    #
     # Return an integer of the result
     def bytes
-      members.reduce(0) { |sum, gf| sum + gf.content.size.to_i }
-    end
+      query = "has_model_ssim:#{file_model}"
+      # Cribbed from Hydra::Collections to speed up the resolution
+      # process
+      args = {
+        fq: "{!join from=hasCollectionMember_ssim to=id}id:#{self.id}",
+        fl: "id, #{file_size_field}",
+        rows: members.count
+      }
+
+      files = ActiveFedora::SolrService.query(query, args)
+      files.reduce(0) { |sum, f| sum += f[file_size_field].to_i }
+   end
+
+   # Field to look up when locating the size of each file in Solr.
+   # Override for your own installation if using something different
+   def file_size_field
+     Solrizer.solr_name('file_size', stored_integer_descriptor)
+   end
+
+   # Override if you are storing your file size in a different way
+   def stored_integer_descriptor
+     Solrizer::Descriptor.new(:integer, :indexed, :stored)
```

Comment:
Would it ever be useful to sort files by size or otherwise filter on the value? This might be moot with the eventual move to PCDM in which case it makes no sense to index it at all and I can just make it a constant.